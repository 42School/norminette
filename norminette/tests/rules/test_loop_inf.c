void	test(void)
{
	v = v[0] * v;
	r[0][0] = m1[0][0] * m2[0][0] + m1[0][1] * m2[1][0] + m1[0][2] * m2[2][0];
}
