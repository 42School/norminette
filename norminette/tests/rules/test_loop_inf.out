[36mtest_loop_inf.c[0m - [32mIsFuncDeclaration[0m In "GlobalScope" from "None" line 1":
		<VOID> <TAB> <IDENTIFIER=test> <LPARENTHESIS> <VOID> <RPARENTHESIS> <NEWLINE>
[36mtest_loop_inf.c[0m - [32mIsBlockStart[0m In "Function" from "GlobalScope" line 2":
		<LBRACE> <NEWLINE>
[36mtest_loop_inf.c[0m - [32mIsAssignation[0m In "Function" from "GlobalScope" line 3":
		<TAB> <IDENTIFIER=v> <SPACE> <ASSIGN> <SPACE> <IDENTIFIER=v> <LBRACKET> <CONSTANT=0> <RBRACKET> <SPACE> <MULT> <SPACE> <IDENTIFIER=v> <SEMI_COLON> <NEWLINE>
[36mtest_loop_inf.c[0m - [32mIsAssignation[0m In "Function" from "GlobalScope" line 4":
		<TAB> <IDENTIFIER=r> <LBRACKET> <CONSTANT=0> <RBRACKET> <LBRACKET> <CONSTANT=0> <RBRACKET> <SPACE> <ASSIGN> <SPACE> <IDENTIFIER=m1> <LBRACKET> <CONSTANT=0> <RBRACKET> <LBRACKET> <CONSTANT=0> <RBRACKET> <SPACE> <MULT> <SPACE> <IDENTIFIER=m2> <LBRACKET> <CONSTANT=0> <RBRACKET> <LBRACKET> <CONSTANT=0> <RBRACKET> <SPACE> <PLUS> <SPACE> <IDENTIFIER=m1> <LBRACKET> <CONSTANT=0> <RBRACKET> <LBRACKET> <CONSTANT=1> <RBRACKET> <SPACE> <MULT> <SPACE> <IDENTIFIER=m2> <LBRACKET> <CONSTANT=1> <RBRACKET> <LBRACKET> <CONSTANT=0> <RBRACKET> <SPACE> <PLUS> <SPACE> <IDENTIFIER=m1> <LBRACKET> <CONSTANT=0> <RBRACKET> <LBRACKET> <CONSTANT=2> <RBRACKET> <SPACE> <MULT> <SPACE> <IDENTIFIER=m2> <LBRACKET> <CONSTANT=2> <RBRACKET> <LBRACKET> <CONSTANT=0> <RBRACKET> <SEMI_COLON> <NEWLINE>
[36mtest_loop_inf.c[0m - [32mIsBlockEnd[0m In "Function" from "GlobalScope" line 5":
		<RBRACE> <NEWLINE>
test_loop_inf.c: OK!
